### DISCUSSION TITLE
Extending a resource - How to reuse decorator arguments

### DISCUSSION THREAD

**Discussion by slamer59** [dagster-io/dagster#9389] [Q&A] [answered]
Hello,
follwogin [this discussion](https://github.com/dagster-io/dagster/discussions/9214) I would like to reuse arguments of the decorator to not copy/paste (and follow updates if any). 

Let's say I have a function that return a function like : 
```@resource( {"some_arg": Field(bool, is_required=False, default_value=True)})
def first_resource(context):
      ... 
```

and the second is using it : 
```
@resource( {"some_arg_2": Field(int, is_required=False, default_value=True), "some_arg": Field(bool, is_required=False, default_value=True)})
def second_resource(context):
      subcontext_for_r1 = context['some_arg_2']
      r1 = first_resource(some_context)
      ... 
```

How would you do to not repeat ?

I would try introspection but feels a bit complex.

**Comment by slamer59** [comment]
Digging in the documentation, I found : _build_init_resource_context_ 
I will try

https://docs.dagster.io/concepts/resources#using-resources-with-ops

**Comment by sryza** [comment, marked as answer]
Another option would be resource-to-resource dependencies: https://docs.dagster.io/concepts/resources#resource-to-resource-dependencies

**Reply by slamer59** [reply]
This is the option I took finally for my use case. Thanks ! There is always a good option in dagster :) just need a bit of reading :)
